src/els_dt_documents.erl

```diff
@@ -0,0 +1,83 @@
+%%==============================================================================
+%% The 'documents' table
+%%==============================================================================
+
+-module(els_dt_documents).
+
+%%==============================================================================
+%% Behaviour els_db_table
+%%==============================================================================
+
+-export([ name/0
+        , opts/0
+        ]).
+
+%%==============================================================================
+%% API
+%%==============================================================================
+
+-export([ insert/1
+        , lookup/1
+        ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% Item Definition
+%%==============================================================================
+
+-record(els_dt_documents, { uri      :: uri()  | '_'
+                          , document :: els_document:document()
+                          }).
+-type els_dt_documents() :: #els_dt_documents{}.
+
+-type item() :: #{ uri      := uri()
```

Comment:
The document already contains the `uri`, why not just have the following?
```
-type item() :: els_document:document().
```

Reply:
That's exactly where I'm heading to in the follow-up commits (not pushed yet).

Reply:
I ended up merging the `els_document` and `els_dt_document` entities into one. I also included the `kind` in the document and got rid of the `modules` table.